### PR TITLE
Removed focus-visible polyfill from menu component

### PR DESCRIPTION
**Pull request by oelhanafey**
@grant-cleary brought up that the focus-visible polyfill is causing performance issues by dispatching too many `visibilitychange` events. The polyfill was used since old versions of Safari do not support the `focus-visible` pseudo-selector. As of Safari 15.4 (March 2022 release), `focus-visible` is now supported.

This PR removes the `focus-visible` polyfill and replaces it with the pseudo-selector. This will cause focus issues with keyboard navigation on older version of Safari. The number of people affected by this change would be relatively small and will continue to decrease, so we decided that it was worth the tradeoff for performance improvements.

### CODE REVIEW

File: components/menu/menu-item-styles.js [hunk 1]
```diff
@@ -27,7 +25,23 @@ export const menuItemStyles = css`
 		z-index: 2;
 	}
 
-	:host([disabled]), :host([disabled]:hover), :host([disabled].focus-visible) {
+	/** separated because safari is having trouble parsing these */
```
Comment (dlockhart) [inline components/menu/menu-item-styles.js:28]: Maybe mention in these comments that it's just Safari <15.4 that has trouble with these entries? That way when/if we eventually stop caring about those versions we can decide to re-combine them.

**Review by dlockhart — APPROVED** [review]
LGTM!

File: components/menu/menu-item-styles.js (hunk 1 above)
Comment (oelhanafey) [reply components/menu/menu-item-styles.js:28]: Yep, good point!

**Review by grant-cleary — APPROVED** [review]
Thanks much, Oscar! :)

**Comment by eKoopmans** [comment]
Associated Rally:
https://rally1.rallydev.com/#/?detail=/userstory/643401699741&fdp=true

**Pull request merged**